frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx

```diff
@@ -0,0 +1,212 @@
+      shouldRenderUnpaidBanner,
+      shouldRenderDatabasePromptBanner,
+    }),
+    withRouter: true,
+  });
+}
+
+function createStateForConfig({
+  shouldRenderPastDueBanner,
+  shouldRenderUnpaidBanner,
+  shouldRenderDatabasePromptBanner,
+}: SetupOpts): State {
+  let isAdmin = false;
+  const tokenStatus = {
+    status: "Token is valid.",
+  };
+  const features = {
+    sso: false,
+  };
+
+  if (shouldRenderPastDueBanner) {
+    tokenStatus.status = "past-due";
+    isAdmin = true;
+  }
+
+  if (shouldRenderUnpaidBanner) {
+    tokenStatus.status = "unpaid";
+    isAdmin = true;
+  }
+
+  if (shouldRenderDatabasePromptBanner) {
+    isAdmin = true;
+    features.sso = true;
+    setupDatabasesEndpoints([createSampleDatabase()]);
+  }
+
+  const state = createMockState({
+    currentUser: createMockUser({ is_superuser: isAdmin }),
+    settings: createMockSettingsState({
+      "token-features": createMockTokenFeatures(features),
+      "token-status": tokenStatus,
+    }),
+  });
+
+  return state;
+}
+
+describe("AppBanner", () => {
+  it.each([
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: false,
+      renderingBanner: "Nothing",
+    },
+    {
+      shouldRenderPastDueBanner: true,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: false,
+      renderingBanner: "Past-due banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: true,
+      shouldRenderDatabasePromptBanner: false,
+      renderingBanner: "Unpaid banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Database prompt banner",
+    },
+    {
+      shouldRenderPastDueBanner: true,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Past-due banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: true,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Unpaid banner",
+    },
+  ] as const)(
+    "should render $renderingBanner when should render past-due banner: $shouldRenderPastDueBanner, should render unpaid banner: $shouldRenderUnpaidBanner, should render database prompt banner: $shouldRenderDatabasePromptBanner",
+    async ({
+      shouldRenderPastDueBanner,
+      shouldRenderUnpaidBanner,
+      shouldRenderDatabasePromptBanner,
+      renderingBanner,
+    }) => {
+      setup({
+        shouldRenderPastDueBanner,
+        shouldRenderUnpaidBanner,
+        shouldRenderDatabasePromptBanner,
+      });
+
+      await assertBanner(renderingBanner);
+    },
+  );
+});
+
+async function assertBanner(
+  renderingBanner:
+    | "Nothing"
+    | "Past-due banner"
+    | "Unpaid banner"
+    | "Database prompt banner",
+) {
+  if (renderingBanner === "Nothing") {
```

Comment:
nit: You can get rid of this method if you split the `it.each` clause into 3 different clauses. Currently this is a bit hard to follow

Reply:
I agree this is too hard to follow.